ccm_web/client/src/utils/canvasSectionNameValidator.ts

```diff
@@ -0,0 +1,55 @@
+import { getCourseSections } from '../api'
+import { CanvasCourseSection } from '../models/canvas'
+import { Course } from '../models/models'
+
+export interface ICanvasSectionNameInvalidError {
+  reason: string
+}
+
+interface IValidator {
+  validate: (sections: CanvasCourseSection[], sectionName: string) => ICanvasSectionNameInvalidError|undefined
+}
+
+class DuplicateSectionNameValidator implements IValidator {
+  validate = (sections: CanvasCourseSection[], sectionName: string): ICanvasSectionNameInvalidError | undefined => {
+    if (sections.map(s => s.name).filter(n => n.localeCompare(sectionName) === 0).length > 0) {
+      return { reason: `Section name already used in this course: "${sectionName}"` }
+    }
+    return undefined
+  }
+}
+
+export class CanvaCoursesSectionNameValidator {
```

Comment:
@chrisrrowland This is nice addition, that you are checking if the section name you created already exist or not. I did not think about this use case.

Reply:
And I misspelled it 🤦‍♂️ 

If there are other client side validations to be done  they should be easy to add.